HLTrigger/Egamma/plugins/HLTScoutingEgammaProducer.cc

```diff
@@ -284,7 +288,13 @@ void HLTScoutingEgammaProducer::produce(edm::StreamID sid, edm::Event& iEvent, e
       if (scRefFromTrk == scRef) {
         d0 = track.d0();
         dz = track.dz();
+        trackpt = track.pt();
+        tracketa = track.eta();
+        trackphi = track.phi();
         charge = track.charge();
+        auto const trackndof = track.ndof();
+        if (trackndof != 0)
+          trackchi2overndf = track.chi2() / trackndof;
       }
     }
```

Comment:
You can probably more conveniently break if you enter this `if`:
```suggestion
      break;
    }
```

Reply:
> I am planning to change all the track parameters to vectors.

I'm no expert (and maybe you already know), but remember that this needs to be done carefully, to ensure backward compatibility (for reading older RAW files in ROOT format, as discussed above). If this change applies to existing variables, changing them to a vector likely breaks backward compatibility (e.g. changing `d0_` from `float` to `std::vector<float>`). _Maybe_ a possible solution would be to remove the "old" variable (e.g. `float d0_`), and add a new one with a different name (e.g. `std::vector<float> trackd0_;`). In any case, it needs to be carefully tested, and bckwd-compatibily must be ensured.

Reply:
@missirol Thank you for the suggestion. I was wondering about the same thing. I did test my changes thoroughly to work on DIGI-RAW data. Can you please suggest whom to contact for backward compatibility? Also, please note that the class version gets updated from 5 to 6 in the `DataFormats/Scouting/src/classes_def.xml` file. Maybe this has some additional functionality?

Reply:
> _Maybe_ a possible solution would be to remove the "old" variable (e.g. `float d0_`), and add a new one with a different name (e.g. `std::vector<float> trackd0_;`).

When reading the 2022 scouting data with this PR, would it be acceptable to just lose the `d0` and `dz` information?

Reply:
Probably not, I overlooked this (thanks). @arsahasransu , please take this into account.

Reply:
@makortel No, we can lose the d0 and the dz information in the collected data.